### PR TITLE
Issue #16: accept arbitrary buffer-compatible objects

### PR DESCRIPTION
**Pull request by pitrou**
This removes hand-written checks for bytes and bytearray, and instead uses the buffer API which allows for arbitrary buffer-compatible types. This is similar to what standard library modules like `zlib` already do.

Incidentally, this also makes it possible to pass in a Numpy array without copying it to a bytes object first (see #35).

### CODE REVIEW

File: lz4/block/_block.c [hunk 1]
```diff
@@ -114,46 +116,29 @@ compress (PyObject * Py_UNUSED (self), PyObject * args, PyObject * kwargs)
...
 #endif
 
+  source_size = (int) source.len;
+  if (source.len != (Py_ssize_t) source_size)
+    {
+      PyBuffer_Release(&source);
+      PyErr_Format(PyExc_OverflowError, "input too large for C 'int'");
+      return NULL;
+    }
+
```
Comment (jonathanunderwood) [inline lz4/block/_block.c]: Pedantic: please be consistent with error message capitalization - "Input.."

**Comment by pitrou** [comment]
You're welcome. I think I've now made the changes you asked for.

**Comment by jonathanunderwood** [comment]
Also, would be super awesome if you could rebase it on top of the development branch.

**Comment by pitrou** [comment]
Ok, I think I've rebased now (actually merged).

**Comment by pitrou** [comment]
Looks like format `p` isn't accepted in PyArg_ParseTuple in Python 3.2 (see Travis-Ci failure), is this something you already noticed?

**Comment by pitrou** [comment]
(if I may suggest so, you could simply drop Python 3.2 or even 3.3 compatibility, I don't think anybody would notice :-))

**Comment by jonathanunderwood** [comment]
Yup - had noticed, just hadn't gotten around to removing 3.2 from the travis and appveyor configs :) Thanks again for your work on this.

**Pull request merged**